components/remote_settings/src/client.rs

```diff
@@ -450,6 +436,83 @@ impl Client {
     }
 }
 
+/// Stores all the endpoints for a Remote Settings server
+///
+/// There's actually not to many of these, so we can just pack them all into a struct
+struct RemoteSettingsEndpoints {
+    /// Top-level URL for Remote Settings server
+    ///
+    /// This has the form `https://[domain]/v1/`. It's where we get the attachment base url from.
+    server_url: Url,
+    /// URL for the collections endpoint
+    ///
+    /// This has the form:
+    /// `https://[domain]/v1/buckets/[bucket-name]/collections/[collection-name]`.
+    ///
+    /// It can be used to fetch some metadata about the collection, but the real reason we use it
+    /// is to get a URL that uniquely identifies the server + bucket name.  This is used by the
+    /// [Storage] component to know when to throw away cached records because the user has changed
+    /// one of these,
+    collection_url: Url,
+    /// URL for the changeset request
+    ///
+    /// This has the form:
+    /// `https://[domain]/v1/buckets/[bucket-name]/collections/[collection-name]/changeset`.
+    ///
+    /// This is the URL for fetching records and changes to records
+    changeset_url: Url,
+    /// URL for the records request
+    ///
+    /// This has the form:
+    /// `https://[domain]/v1/buckets/[bucket-name]/collections/[collection-name]/records`.
+    ///
+    /// This is the old/deprecated way to get records
+    records_url: Url,
+}
+
+impl RemoteSettingsEndpoints {
+    /// Construct a new RemoteSettingsEndpoints
+    ///
+    /// `base_url` should have the form `https://[domain]/v1` (no trailing slash).
+    fn new(base_url: &Url, bucket_name: &str, collection_name: &str) -> Result<Self> {
+        let mut server_url = base_url.clone();
+        // Push the empty string to add a trailing slash.  This avoids a redirect when we hit
+        // the server.
+        Self::path_segments_mut(&mut server_url)?.push("");
```

Comment:
Using `https://firefox.settings.services.mozilla.com/v1/` instead of `https://firefox.settings.services.mozilla.com/v1` seems right to me.  I always try to avoid hitting a redirect when I make an API call.  But tell me if that's a bad idea.

Reply:
There should not be any redirect, we don't call the `server_url` directly (without trailing slash)